### PR TITLE
Update Doctrine DBAL to V3

### LINKED ISSUE #10613: [Improvement]: Doctrine  DBAL v3
Should fix #5361 as well. 
Consider discussion here: #10648

See also: #12917

### PR DESCRIPTION
**Pull request by blankse**
Resolve https://github.com/pimcore/pimcore/issues/10613

### CODE REVIEW

**Comment by aashan10** [comment]
@blankse please fix the merge conflicts

File: lib/Model/Listing/AbstractListing.php [hunk 1]
```diff
@@ -44,9 +44,9 @@ abstract class AbstractListing extends AbstractModel implements \Iterator, \Coun
     protected $limit;
 
     /**
-     * @var int|null
+     * @var int
      */
-    protected $offset;
+    protected $offset = 0;
```
Comment (aashan10) [inline lib/Model/Listing/AbstractListing.php:49]: @brusch does this fall under the backwards compatibility promise that we discussed earlier or not? This seems to be for pimcore 11, so I am a bit confused!

**Review by aashan10 — APPROVED** [review]

File: lib/Model/Listing/AbstractListing.php (hunk 1 above)
Comment (dvesh3) [reply lib/Model/Listing/AbstractListing.php:49]: We can introduce breaking changes since this is for Pimcore 11.

**Comment by dvesh3** [comment]
@blankse thank you very much!

**Comment by NiklasBr** [comment]
Great work! In a few days you do it again for version 4 😉 

https://github.com/doctrine/dbal/milestone/3

**Comment by brusch** [comment]
We should definitely prepare already for 4 and check that we're not using any deprecated methods: 
https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md

**Comment by brusch** [comment]
Created a follow-up ticket: #13015

**Pull request merged**